src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.cs

```diff
@@ -89,9 +94,19 @@ public static SslStreamCertificateContext Create(X509Certificate2 target, X509Ce
                 }
             }
 
-            return new SslStreamCertificateContext(target, intermediates, trust);
+            SslStreamCertificateContext ctx = new SslStreamCertificateContext(target, intermediates, trust);
+
+            // On Linux, AddRootCertificate will start a background download of an OCSP response,
+            // unless this context was built "offline".
```

Comment:
Note this will happen for both client & server certificates. While we do not expose the option publicly, we convert all certificates to context to simplify later processing. 
I think it is one to staple responses to server but I'm not sure if this is what you want.

Reply:
Well, assuming the protocol allows for OCSP stapling with client auth certs there's goodness here... but I certainly didn't test it and there's probably more work involved in it; so I'll find a way to turn it off.

Reply:
On client, it is only called here: 
https://github.com/dotnet/runtime/blob/34367585eba1d1a169dce2c97d7a54592cdc755e/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Protocol.cs#L548

We can add optional parameter for internal constructor to indicate client side. (at least for now)